kv/kvclient/kvcoord: TestDistSenderRangeFeedRetryOnTransportErrors failed

kv/kvclient/kvcoord.TestDistSenderRangeFeedRetryOnTransportErrors [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10716267?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10716267?buildTab=artifacts#/) on release-23.1 @ [a5b5bd2d06035e75ed7ef9498120e1426bc6a8f3](https://github.com/cockroachdb/cockroach/commits/a5b5bd2d06035e75ed7ef9498120e1426bc6a8f3):


```
=== RUN   TestDistSenderRangeFeedRetryOnTransportErrors
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/34b0380b5c9a98f3e962b8dbf4ed4846/logTestDistSenderRangeFeedRetryOnTransportErrors183913398
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestDistSenderRangeFeedRetryOnTransportErrors
    dist_sender_rangefeed_mock_test.go:182: -- test log scope end --
--- FAIL: TestDistSenderRangeFeedRetryOnTransportErrors (1.41s)
=== RUN   TestDistSenderRangeFeedRetryOnTransportErrors/mux=true/FailedPrecondition
    dist_sender_mux_rangefeed.go:277: Unexpected call to *kvcoord.MockTransport.IsExhausted([]) at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go:277 because: 
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:107 has already been called the max number of times
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:107 has already been called the max number of times
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:112 has already been called the max number of times
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:121 has already been called the max number of times
        expected call at github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:147 has already been called the max number of times
    controller.go:137: missing call(s) to *kvpbmock.MockInternal_MuxRangeFeedClient.Send(is anything) github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go:136
    controller.go:137: aborting test due to missing call(s)
    --- FAIL: TestDistSenderRangeFeedRetryOnTransportErrors/mux=true/FailedPrecondition (0.36s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/replication
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDistSenderRangeFeedRetryOnTransportErrors.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29171

--- a/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go
+++ b/pkg/kv/kvclient/kvcoord/dist_sender_rangefeed_mock_test.go
@@ -144,9 +144,9 @@ func TestDistSenderRangeFeedRetryOnTransportErrors(t *testing.T) {
 							client.EXPECT().RangeFeed(gomock.Any(), gomock.Any()).Return(stream, nil)
 						}
 
-						transport.EXPECT().IsExhausted().Return(false)
-						transport.EXPECT().NextReplica().Return(desc.InternalReplicas[0])
-						transport.EXPECT().NextInternalClient(gomock.Any()).Return(client, nil)
+						transport.EXPECT().IsExhausted().Return(false).AnyTimes()
+						transport.EXPECT().NextReplica().Return(desc.InternalReplicas[0]).AnyTimes()
+						transport.EXPECT().NextInternalClient(gomock.Any()).Return(client, nil).AnyTimes()
 						transport.EXPECT().Release().AnyTimes()
 					}
 

